acceptance/cluster/localcluster.go

```diff
@@ -504,11 +504,18 @@ func (l *LocalCluster) Start() {
 func (l *LocalCluster) Assert(t *testing.T) {
 	const almostZero = 50 * time.Millisecond
 	filter := func(ch chan Event, wait time.Duration) *Event {
+		waitTimer := time.NewTimer(wait)
+		waitTimerRead := false
 		for {
+			if !waitTimer.Reset(wait) && !waitTimerRead {
+				<-waitTimer.C
+			}
+			waitTimerRead = false
 			select {
 			case act := <-ch:
 				return &act
-			case <-time.After(wait):
+			case <-waitTimer.C:
+				waitTimerRead = true
```

Comment:
I think it is worthwhile to put this pattern in `util`:

```
type Timer struct {
  *time.Timer
  Read bool
}

// Reset operates on a value so that Timer can be stack allocated.
func (t Timer) Reset(wait time.Duration) Timer {
  if t.Timer == nil {
    t.Timer = time.NewTimer(wait)
    return t
  }
  if !t.Timer.Reset(wait) && !t.Read {
    <-t.Timer.C
  }
  t.Read = false
  return t
}
```

Then, to use:

```
  var timer util.Timer
  for {
    timer = timer.Reset(wait)
    switch {
    case <-timer.C:
      timer.Read = true
      ...
    }
  }
```

Still irritating that you have to note when the timer channel has been read, but I can't think of anything better right now. This would also avoid creating and then immediately resetting the timer (though I doubt that is any sort of performance bottleneck).